Neos.Fusion/Resources/Private/Fusion/Root.fusion

```diff
@@ -289,3 +290,76 @@ prototype(Neos.Fusion:GlobalCacheIdentifiers) < prototype(Neos.Fusion:DataStruct
   format = ${request.format}
   baseUri = ${String.toString(BaseUri.getConfiguredBaseUriOrFallbackToCurrentRequest(request.httpRequest))}
 }
+
+# returns the current flashMessages as array, passes then to the renderer or to the content
+#
+# Usage:
+# flashMessages = Neos.Fusion:FlashMessages
+# renderer = afx`
+#   <Neos.Fusion:Loop items={props.flashMessages}>
+#     {item.message}
+#   </Neos.Fusion:Loop>
+# `
+#
+# Usage with renderer:
+# renderedFlashMessages = Neos.Fusion:FlashMessages {
+#   renderer = My.Package:FlashMessageRenderer
+# }
+# renderer = afx`
+#   {props.renderedFlashMessages}
+# `
+#
+# AFX only:
+# renderer = afx`
+#   <Neos.Fusion:FlashMessages>
+#     <Neos.Fusion:Loop items={flashMessages}>
+#       {item.message}
+#     </Neos.Fusion:Loop>
+#   </Neos.Fusion:FlashMessages>
+# `
+prototype(Neos.Fusion:FlashMessages) {
+  @class = 'Neos\\Fusion\\FusionObjects\\FlashMessagesImplementation'
+  @sortProperties = false
+
+  # severity of the messages to return (One of the Neos\Error\Messages\Message::SEVERITY_* constants)
+  severity = null
+
+  # whether the messages should be flushed after reading them, default is true
+  flushMessages = true
+
+  # name of the property that contains the flashMessages inside the prototype
+  collectionName = 'flashMessages'
+
+  # add flashMessages in the fusion code, expects an array of Neos.Fusion:FlashMessages.Message elements
+  additionalMessages = Neos.Fusion:DataStructure
+}
+
+# to pass additional flashMessages to Neos.Fusion:FlashMessages within the fusion code
+#
+# Usage:
+# flashMessages = Neos.Fusion:FlashMessages {
+#   additionalMessages {
+#     testCase1 = Neos.Fusion:FlashMessages.Message {
+#       message = 'Test'
+#     }
+#   }
+# }
+prototype(Neos.Fusion:FlashMessages.Message) < prototype(Neos.Fusion:DataStructure) {
```

Comment:
i was wondering what the usecase for additionalMessages is ... if this is only for functional testing - we can easily mock the flashmessage service ;)

Reply:
the use case for this would not be functional testing, but testing the output of flashmessages in the presentation layer. e.g. with Sitegeist.Monocle, like
```
@styleguide {
  props {
    flashMessages {
      additionalMessages {
        testCase1 = Neos.Fusion:FlashMessages.Message {
        message = 'Test'
      }
    }
  }
}
```

Reply:
hmm - i would be in the opinion that a flashmessage dto should not reach the presentation layer (same like node) and be mapped to an array. That way we dont need to instantiate the objects for monocle and the component can render independent of the message dto

Reply:
also it reduces complexity for that we dont need the Flashmessages object and no additional flashmessages logic